sdk/src/transport.rs

```diff
@@ -1,20 +1,15 @@
 use crate::transaction::TransactionError;
-use std::{error, fmt, io};
+use std::io;
+use thiserror::Error;
 
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum TransportError {
-    IoError(io::Error),
-    TransactionError(TransactionError),
-}
-
-impl error::Error for TransportError {}
-impl fmt::Display for TransportError {
-    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
-        match self {
-            TransportError::IoError(err) => write!(formatter, "{:?}", err),
-            TransportError::TransactionError(err) => write!(formatter, "{:?}", err),
-        }
-    }
+    #[error("transport io error: {0}")]
+    IoError(#[from] io::Error),
+    #[error("transport transaction error: {0}")]
+    TransactionError(#[from] TransactionError),
+    #[error("transport custom error: {0}")]
+    Custom(String),
```

Comment:
Might kill the `String` here in favor of https://crates.io/crates/anyhow if time permits.  Would allow removal of its companion variant in `ClientError`

Reply:
Tough call.  `anyhow` is another escape hatch.  The team tends to make use of those too often.

Reply:
Yeah I was waffling on that notion.  Probably better to yank the dependency-based variants out of `ClientError` in favor of our own variants and some `From` impls